### ISSUE TITLE
Budgets for which the source account is a debt aren't consistently classified as spending

### ISSUE THREAD

**Issue by notinecrafter** [firefly-iii/firefly-iii#6453]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I am the creditor on a debt, which regularly increases via transactions from my asset account.
These transactions are modeled as a transaction from my asset account to the debt, which works quite well.

I also keep track of what sort of spending I do on this debt by sorting them into budgets (which only contain transactions that go towards this debt).
When in the withdrawals page, I can see these transactions as counting towards spending on a budget, because they go out of my asset account.
In the budgets page, however, I see all my budgets as having 0 spending, because the transactions don't lower my net worth.

### Expected behaviour

Either the transactions count for the budgets in the budgets page *and* the withdrawals page, or they count for neither.

### Debug information

Debug information generated at 2022-09-21 22:54:13 Etc/UTC for Firefly III version **5.7.10**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.10 |
| Firefly III API | 1.5.6 |
| Build | (unknown), (unknown) |
| DB version | 18 (exp. 18) |
| PHP | `8.1.9` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Etc/UTC |
| Browser TZ | UTC |
| App environment | local |
| App debug mode | true |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql |
| Current driver | mysql |
| Trusted proxies (.env) | `` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-09-01 00:00:00 |
| Session end | 2022-09-30 23:59:59 |
| Session first | 2022-05-01 00:00:00 |
| User ID | 1 |
| User language | en_GB |
| User locale | en_GB |
| Attempt at "en_GB.utf8" | 'en_GB.utf8' |
| Attempt at "en_GB.UTF-8" | 'en_GB.UTF-8' |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:104.0) Gecko/20100101 Firefox/104.0 |

### Steps to reproduce

1. Create a debt which you are owed
2. Create a transaction from your asset account to this debt
3. Assign a budget to this debt
4. Observe that this transaction contributes to the budgets pie-chart in the withdrawals page
5. Observe that this transaction does not contribute to the spending in the budget
6. Ponder the inconsistency

### Additional info

Ideally, I would suggest solving this by making a setting on the liability whether people want transactions of the aforementioned kind to count towards the budget or not. 
Even if the above inconsistency is actually intended behaviour, this solution would still be a nice feature for people who don't want said behaviour.

I'm willing to work on a pull request that adds this functionality if you agree with my suggestion.

**Comment by JC5** [comment]
Where exactly "in the budget" is it not counted?

**Comment by notinecrafter** [comment]
In the budgets page, transactions which are assigned to a budget, but which go from an asset account to a liability do not detract from the amount spent in that budget.

**Comment by notinecrafter** [comment]
<img width="1193" alt="Screenshot 2022-09-22 at 19 14 50" src="https://user-images.githubusercontent.com/22083088/191832565-ea9736ae-c43f-4a8c-870b-58debb4acf3c.png">

Basically, the above

**Comment by JC5** [comment]
Ah, nice find. I'll pick it up :+1:

**Comment by JC5** [comment]
Ah I saw the code and now I remember.
```
        // this collector excludes all transfers TO
        // liabilities (which are also withdrawals)
        // because those expenses only become expenses
        // once they move from the liability to the friend.
```

**Comment by notinecrafter** [comment]
@JC5 Thank you for checking it out so quickly - great customer service is a very nice feature in FOSS :)

Unfortunately, I don't think the story of the code comment is exactly what's happening in my case. As I understand it, the comment accounts for the following scenario:

1. A friend buys me a cookie
2. I register this as an expense, but instead of taking the money from my bank account I take it from my debt to the friend. Because I've not actually spent any money, it doesn't need to be budgeted.
3. As soon as I repay my friend, the money that I actually spent can be budgetted.

However, I do not owe this debt; I am owed this debt. As such, the following is happening:

1. I buy my friend a cookie
2. I enter this as an expense to my friends debt - so I know they'll have to pay me back later. I'd like to budget this money, as I have spent it
3. As soon as my friend pays me back - which in my case, can take a while - I get money back, essentially undoing the expense.

In other words instead of what the comment says - the expense only becomes an expense when it is moved from the liability to the friend - it's almost the opposite, where the expense is an expense *until* the money moves from my friend into the liability.

If the term "collector" implies the infrastructure that I think it implies, you could probably fix this by only excluding liabilities which one is owes, and not those which one is owed.

**Comment by JC5** [comment]
If you buy something for your friend, use the liability as a source. Then it will be properly budgeted.

**Comment by JC5** [comment]
The amount due will increase, and the balance will decrease. Ie: the balance will dip back to zero or below it, and the amount due (to you) will increase.

A debt owed to you should have a balance of zero, and a positive amount due. If the balance is positive, there is money paid back that you are free to use: transfer it to your asset account and then spend it yourself. The "transfer" to your asset account will be labeled a deposit.
